OpenOversight/app/main/views.py

```diff
@@ -565,6 +565,85 @@ def submit_department_images(department_id=1):
+
+
+@main.route('/download/department/<int:department_id>', methods=['GET'])
+@limiter.limit('5/minute')
+def download_dept_csv(department_id):
+    department = Department.query.filter_by(id=department_id).first()
+    records = Officer.query.filter_by(department_id=department_id).all()
+    if not department or not records:
+        abort(404)
+    dept_name = records[0].department.name.replace(" ", "_")
+    first_row = "id, last, first, middle, suffix, gender, "\
+                "race, born, employment_date, assignments, notes\n"
+
+    assign_dict = {}
+    assign_records = Assignment.query.all()
+    for r in assign_records:
+        if r.officer_id not in assign_dict:
+            assign_dict[r.officer_id] = []
+        assign_dict[r.officer_id].append("(#%s %s %s %s %s)" % (check_input(r.star_no), check_input(r.rank), check_input(r.unit), check_input(r.star_date), check_input(r.resign_date)))
+
+    notes_dict = {}
+    notes_records = Note.query.all()
+    for n in notes_records:
+        if n.officer_id not in notes_dict:
+            notes_dict[n.officer_id] = []
+        notes_dict[n.officer_id].append("(%s %s %s)" % (check_input(n.note), check_input(n.date_created), check_input(n.date_updated)))
+
+    record_list = ["%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s\n" %
+                   (str(record.id),
+                    check_input(record.last_name),
+                    check_input(record.first_name),
+                    check_input(record.middle_initial),
+                    check_input(record.suffix),
+                    check_input(record.gender),
+                    check_input(record.race),
+                    check_input(record.birth_year),
+                    check_input(record.employment_date),
+                    " ".join(assign_dict.get(record.id, [])),
+                    " ".join(notes_dict.get(record.id, []))
+                    ) for record in records]
+
+    csv_name = dept_name + "_Officers.csv"
+    csv = first_row + "".join(record_list)
+    csv_headers = {"Content-disposition": "attachment; filename=" + csv_name}
+    return Response(csv, mimetype="text/csv", headers=csv_headers)
+
+
+@main.route('/download/department/<int:department_id>/incidents', methods=['GET'])
+@limiter.limit('5/minute')
+def download_incidents_csv(department_id):
+    department = Department.query.filter_by(id=department_id).first()
+    records = Incident.query.filter_by(department_id=department.id).all()
+    if not department or not records:
+        abort(404)
+    dept_name = records[0].department.name.replace(" ", "_")
+    first_row = "id,report_num,date,description,location,licences,links,officers\n"
+
+    record_list = ["%s,%s,%s,%s,%s,%s,%s,%s\n" %
+                   (str(record.id),
+                    check_input(record.report_number),
+                    check_input(record.date),
+                    check_input(record.description),
+                    check_input(record.address),
```

Comment:
Let's add a string representation for `Location` so the output here is nice, i.e.:

```
diff --git a/OpenOversight/app/models.py b/OpenOversight/app/models.py
index c4b944b..9e87a13 100644
--- a/OpenOversight/app/models.py
+++ b/OpenOversight/app/models.py
@@ -222,6 +222,20 @@ class Location(db.Model):
     def validate_state(self, key, state):
         return state_validator(state)
 
+    def __repr__(self):
+        if self.street_name and self.cross_street2:
+            return 'Intersection of {} and {}, {} {}'.format(
+                self.street_name, self.cross_street2, self.city, self.state)
+        elif self.street_name and self.cross_street1:
+            return 'Intersection of {} and {}, {} {}'.format(
+                self.street_name, self.cross_street1, self.city, self.state)
+        elif self.street_name and self.cross_street1 and self.cross_street2:
+            return 'Intersection of {} between {} and {}, {} {}'.format(
+                self.street_name, self.cross_street1, self.cross_street2,
+                self.city, self.state)
+        else:
+            return '{} {}'.format(self.city, self.state)
+
```